### PR TITLE
feat: xcom bq slot

### PR DESCRIPTION
**Pull request by irainia**
This PR is to add BQ slot usage on the job for monitoring purposes. Changes:
* add callback handler when BQ job finished
* send BQ slot usage through XCOM
* fix minor bug to show error when encountered one during BQ client initializtion

### CODE REVIEW

File: task/bq2bq/executor/main.py [hunk 1]
```diff
@@ -30,4 +25,15 @@
         app_config.dry_run,
         app_config.job_labels,
         app_config.xcom_path,
+        on_finished_job = job_handler.handle_finished_job,
```
Comment (Mryashbhardwaj) [inline task/bq2bq/executor/main.py:28]: can we call this `on_job_finish` ?

Comment (Mryashbhardwaj) [inline task/bq2bq/executor/main.py:28]: can we call this `on_job_finish` ?

Comment (irainia) [reply task/bq2bq/executor/main.py:28]: How about `on_job_finished`? This is to imply that the job did finish in the past and not occurring right now. Similar to why most of the time we encounter `Done` function or method to indicate whether activity already happened or not. An example is like function `Done` in `context` or function `Done` in BigQuery client. Yes, `Do` sometime is used but only when we want to do something that hasn't happened yet.

Comment (Mryashbhardwaj) [reply task/bq2bq/executor/main.py:28]: i think `on_job_finish` is better, as most of the event callbacks are already using similar tense. 
for eg: `on_retry_callback` is not called `on_retried_callback` , thats why

Comment (irainia) [reply task/bq2bq/executor/main.py:28]: sure

Comment (irainia) [reply task/bq2bq/executor/main.py]: updated

Comment (irainia) [reply task/bq2bq/executor/main.py]: updated

**Review by Mryashbhardwaj — APPROVED** [review]

**Pull request merged**